packages/client/hmi-client/src/components/navbar/tera-notification-panel.vue

```diff
@@ -0,0 +1,225 @@
+	switch (item.type) {
+		case ClientEventType.ExtractionPdf:
+			return 'PDF extraction from';
+		default:
+			return 'Process';
+	}
+};
+
+const getActionText = (item: NotificationItem) => {
+	switch (item.type) {
+		case ClientEventType.ExtractionPdf:
+			return 'Extracting...';
+		default:
+			return 'Processing...';
+	}
+};
+
+const getAssetRoute = (item: NotificationItem) => {
+	switch (item.type) {
+		case ClientEventType.ExtractionPdf:
+			return { assetId: item.id, pageType: AssetType.Document };
+		default:
+			return { assetId: item.id, pageType: AssetType.Document };
+	}
+};
+
+const getElapsedTimeText = (item: NotificationItem) => {
+	const time = Date.now() - item.lastUpdated;
+	const minutes = Math.floor(time / (1000 * 60));
+	return minutes > 0 ? `${minutes} minutes ago` : 'Just now';
+};
+</script>
+
+<style>
+.notification-button {
+	position: relative;
+	.p-button.p-button-secondary.p-button-text {
+		color: var(--text-color-subdued);
+	}
+	.p-button.p-button-secondary.p-button-text:enabled:focus,
+	.p-button.p-button-secondary.p-button-text:enabled:hover {
+		background-color: #e3efe4;
+	}
+	.badge {
+		position: absolute;
+		background-color: var(--error-color);
+		color: #fff;
+		top: -5px;
+		right: -5px;
+		font-size: var(--font-tiny);
+		border-radius: 10px;
+		padding: 2px 5px;
+	}
+}
+/* Reset default overlay component style */
+.notification-container.p-overlaypanel .p-overlaypanel-content {
+	padding: 0;
+}
+.notification-container.p-overlaypanel:after,
+.notification-container.p-overlaypanel:before {
+	content: none;
+}
+.notification-container.p-overlaypanel {
+	top: 51px !important;
+	width: 540px;
+	box-shadow: 0px 4px 4px 0px #00000040;
+	padding: var(--content-padding);
+	padding-bottom: 1.5rem;
+	gap: var(--gap);
+	border: 1px solid #c3ccd6;
+	border-radius: var(--border-radius-medium);
+}
+.header {
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+
+	h1 {
+		font-size: var(--font-body-medium);
+		font-weight: var(--font-weight-semibold);
+	}
+	button {
+		padding: 1px;
+	}
+}
+.text-body {
+	font-size: var(--font-body-small);
+	color: var(--text-color-secondary);
+}
+
+.notification-items-container {
+	max-height: 570px;
+	list-style: none;
+	overflow: auto;
+}
+
+.notification-item {
+	padding: 1rem 0;
+	&:not(:first-child) {
+		border-top: 1px solid #dee2e6;
+	}
+	.heading {
+		font-size: var(--font-body-small);
+	}
+	.heading a {
+		color: var(--primary-color);
+	}
+	.msg {
+		font-size: var(--font-caption);
+		color: #9298a5;
+	}
+	.action {
+		font-size: var(--font-caption);
+		color: var(--text-color-secondary);
+	}
+	.progressbar-container {
+		margin-top: 0.5rem;
+		display: flex;
+		justify-content: space-between;
+		align-items: center;
+		gap: var(--gap-small);
+	}
+	.p-progressbar {
+		flex-grow: 1;
+	}
+
+	.done-container {
+		margin-top: 0.5rem;
+		display: flex;
+		justify-content: space-between;
+		align-items: center;
+		gap: var(--gap-small);
+	}
```

Comment:
You can use CSS nesting now, which is really neat.
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_nesting/Using_CSS_nesting#combinators

Reply:
Yeap it's nice to use nesting with native CSS!. It's already one level nested and I just flattened it just to avoid going too deeply nested but I can update if that feels more neater.